### ISSUE TITLE
no augmentation for validation set

### ISSUE THREAD

**Issue by dbuscombe-usgs** [Doodleverse/segmentation_gym#30]
One idea:

create different generators for train and validation sets feed the two generators two different dictionaries. train dictionary takes the augmentation arguments, validation takes no augmentation

Disadvantage: would require train and validation sets to be set by makedatasets, not changed on the fly using a config file, so every time you wanted a different val split you'd have to delete and remake the npz files

Alternative idea:
change the train script to evaluate against a second validation set, made using makedatasets but with no augmentation

Disadvantage: requires the user to run makedatasets twice, or add extra params to the config file and changes to make datasets --- I think the latter is preferable

**Comment by dbuscombe-usgs** [comment]
Trial solution: we make two labels sets, one augmented and one not

In `make_datasets`, because `seed=SEED` and `shuffle=False`, image and mask generators can be zipped to make batches of augmented masks whereby the correct image is always paired

by the same logic, a dictionary of null augmentations can be passed to a separate generator pair zipped up, and spit out the same label-image pairs without augmentations. We'll call this oversampled validation set with no augmentations the 'hold out' or 'test' set, in ML convention

define the null generator
```
null_data_gen_args = dict(featurewise_center=False,
                     featurewise_std_normalization=False,
                     rotation_range=0,
                     width_shift_range=0,
                     height_shift_range=0,
                     fill_mode='reflect', 
                     zoom_range=0,
                     horizontal_flip=False,
                     vertical_flip=False)    
                     
null_mask_datagen = tf.keras.preprocessing.image.ImageDataGenerator(**null_data_gen_args)
```
Make it flow from directory in the exact same way as the image and augmented mask generators

```
null_mask_generator = null_mask_datagen.flow_from_directory(
        lab_path,
        target_size=(NX, NY),
        batch_size=int(n_im/AUG_LOOPS),
        class_mode=None, seed=SEED, shuffle=False, color_mode="grayscale", interpolation="nearest")
```

when in the augmentation loop, 

```
            null_train_generator = (pair for pair in zip(img_generator, null_mask_generator))
            #grab a batch of images and label images
            _, null_y = next(null_train_generator)
```

which gets passed to the for loop like this:

```
for im,lab,null_lab,file in zip(x,y,null_yfilenames):
```

and then parallel operations on `lab` and `null_lab` ensue ...

**Comment by dbuscombe-usgs** [comment]
after much trial, this approach does not work (it shuffles the images)

so, I will create all the necessary inputs for the loop to look like
```
for im,nim,lab,nlab,file,nfile in zip(x,null_x,y,null_y, filenames, null_filenames):
```

and deal with each set of augmented and non-augmented images, labels, and filenames separately ...

**Comment by dbuscombe-usgs** [comment]
success. augmented and non-augmented examples and samples are now made

**Comment by dbuscombe-usgs** [comment]
needs

- [ ] docs

**Comment by dbuscombe-usgs** [comment]
* mention in docs how it is possible to train using non-augmented imagery by making them all Null values in the config

* this will be implemented only to present the option to the user of testing trained models with non-augmented imagery

**Comment by dbuscombe-usgs** [comment]
new `make_nd_dataset.py` now automatically makes one set of "noaug" files

**Comment by dbuscombe-usgs** [comment]
A switch needs to be implemented for users who do not want to use augmented imagery
